projects/etos_suite_runner/src/etos_suite_runner/otel_tracing.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+# Copyright Axis Communications AB.
+#
+# For a full list of individual contributors, please see the commit history.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -*- coding: utf-8 -*-
+"""OpenTelemetry-related code."""
+import logging
+import os
+import opentelemetry
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_current_context() -> opentelemetry.context.context.Context:
+    """Get current context (propagated via environment variable OTEL_CONTEXT)."""
+    carrier = {}
+    LOGGER.info("Current OpenTelemetry context env: %s", os.environ.get("OTEL_CONTEXT"))
+    for kv in os.environ.get("OTEL_CONTEXT", "").split(","):
+        if kv:
+            k, v = kv.split("=", 1)
+            carrier[k] = v
+    ctx = opentelemetry.propagate.extract(carrier)
+    LOGGER.info("Current OpenTelemetry context %s", ctx)
+    return ctx
```

Comment:
We do something similar to this in the ETOS library where we utilize the built-in textmap propagator.
Called here: https://github.com/eiffel-community/etos-library/blob/main/src/etos_lib/eiffel/subscriber.py#L104-L106 
Extracted here: https://github.com/eiffel-community/etos-library/blob/main/src/etos_lib/eiffel/subscriber.py#L37-L52

I believe that the one in ETOS library is simpler.

Reply:
In the latest update I have made it more idiomatic. It does not make the code simpler, but it may look better.